### PR TITLE
Gherkin Scripts Workload Groupings and Resources Drop Down Menu Scenarios - [ODC-2922]

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4073

Problem: Gherkin Scripts Design for Workload Groupings and Resources Drop Down Menu epic ODC-2922

Solution: These are the gherkin scripts designed covering all the scenarios related to Workload Groupings and Resources Drop Down Menu epic ODC-2922

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature [hunk 1]
```diff
@@ -0,0 +1,94 @@
+Feature: Topology
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:1]: Update feature description as "Resources List view in topology"

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature [hunk 2]
```diff
@@ -0,0 +1,94 @@
+Feature: Topology 
+    User will be able to show and hide all types of resources on Topology graph and list view
+
+
+Background:
+    Given user logged into the openshift application
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:6]: We can remove this step, going forward.. As our scripts executes on localhost

**Comment by jeff-phillips-18** [comment]
The resource filter design is still in flux. Last I heard was that they will behave much like the catalog pages where if nothing is selected there are no filters. So, the default would be nothing selected. To show only DCs one would select the DC checkbox. To hide only DCs, the user would need to select everything except the DCs.

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:1]: Changed it to "Resources Groupings in Topology"

**Comment by gajanan-more** [comment]
> Please add @Manual tag only if automation is not possible for that scenarios

Yes @makambalaji I thought about it and then only added the tags. These are the things we cannot determine in the automation. These things must to checked manually. There can be multiple cases of it and it's impossible to automate it.

**Comment by gajanan-more** [comment]
> The resource filter design is still in flux. Last I heard was that they will behave much like the catalog pages where if nothing is selected there are no filters. So, the default would be nothing selected. To show only DCs one would select the DC checkbox. To hide only DCs, the user would need to select everything except the DCs.

I will update the scenarios accordingly @jeff-phillips-18

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:6]: Should we remove the Background from all the features as we can directly mention it in the Given statement? WDYT @makambalaji

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+
+Background:
+    Given user is at developer perspecitve
+
+
+@regression, @smoke
+Scenario: Default state of Resources dropdown
+    Given user is at Topology page
+    When user click on the Resources dropdown
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:12]: *clicks

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+Scenario: Ability to hide Helm Releases resource types in Topology graph and list view
+    Given user is at Topology page
+    When user clicks on the Resources dropdown 
+    And user checks all checkboxes
+    And user unchecks Helm Releases checkbox
+    Then user will not see the workloads of Helm Releases resource types only
+
+
+@regression, @smoke, @manual
+Scenario: Ability to show Services resource types in Topology graph and list view
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature:125]: We don't show services in topology

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Background:
+    Given user is at developer perspecitve
+
+
+@regression, @smoke
+Scenario: Default state of Display dropdown
+    Given user is at the Topology page
+    When user clicks on the Display dropdown
+    Then user will see the Show Groups is checked
+    And user will see the Expand is checks
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:14]:
```suggestion
    And user will see the Expand is checked
```

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    And user will see the Helm Releses checkbox checked
+    And user will see the Operator Groups checkbox checked
+
+
+@regression, @smoke
+Scenario: Uncheck the Show Groups
+    Given user is at the Topology page
+    When user clicks on the Display dropdown
+    And user unchecks the Show Groups
+    Then user will see that the Expand and it's children are disabled
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:25]: User will see that group no longer appear in the view, only workloads (both standalone and those belonging to groupings)

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    Then user will see the grouped resources collapsed
+
+
+@regression, @smoke, @manual
+Scenario: Provide ability to hide and show Helm release groupings in Topology graph and list view
+    Given user is at the Topology page
+    And user is at the graph view
+    When user clicks on the Display dropdown
+    And user unchecks the Helm Release checkbox
+    Then user will see the Helm releases ungrouped
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:63]:
```suggestion
    Then user will see the Helm releases collapsed, showing a summary of its workloads 
```

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    Then user will see the Helm releases ungrouped 
+
+
+@regression, @smoke, @manual
+Scenario: Provide ability to hide and show Knative Services groupings in Topology graph and list view
+    Given user is at the Topology page
+    And user is at the graph view
+    When user clicks on the Display dropdown
+    And user unchecks the Knative Services checkbox
+    Then user will see the Knative Services ungrouped
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:72]:
```suggestion
    Then user will see the Knative Services collapsed, showing a summary of its workloads
```

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+
+@regression, @smoke, @manual 
+Scenario: Provide ability to hide and show Operator Groups groupings in Topology graph and list view
+    Given user is at the Topology page
+    And user is at the graph view
+    When user clicks on the Display dropdown
+    And user unchecks the Operator Groups checkbox
+    Then user will see the Operator Groups ungrouped
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:81]:
```suggestion
    Then user will see the Operator Groups collapsed, showing a summary of its workloads
```

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature:25]: Updated the statement

File: frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/workload-Groupings-ODC-2922.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/topology/resources-Groupings-ODC-2922.feature]: Removed from both the files

**Pull request merged**